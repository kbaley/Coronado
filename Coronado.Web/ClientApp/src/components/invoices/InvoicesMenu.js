import React, { Component } from 'react';
import { withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import ListAltIcon from '@material-ui/icons/ListAlt';
import { sumBy } from 'lodash';
import { CurrencyFormat } from '../common/CurrencyFormat';
import { SidebarMenuItem } from '../common/SidebarMenuItem';

export class InvoicesMenu extends Component {
  render() {
    return (
      <SidebarMenuItem 
        to='/invoices' 
        primary="Invoices" 
        icon={<ListAltIcon />}
        secondary={<CurrencyFormat value={sumBy(this.props.invoices, i => { return i.balance })} />}
        />
    );
  }
}

function mapStateToProps(state) {
  return {
    invoices: state.invoices
  }
}

export default withRouter(connect(
  mapStateToProps,
  null,
  null,
  {pure:false}
)(InvoicesMenu));