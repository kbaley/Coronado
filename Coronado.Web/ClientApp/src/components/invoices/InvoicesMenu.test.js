import { InvoicesMenu } from "./InvoicesMenu";
import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from "react-router";

describe('InvoicesMenu tests', () => {
  it('should render an InvoicesMenu', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
        <InvoicesMenu />
        </MemoryRouter>
      );
    expect(tree.toJSON()).toMatchSnapshot();
  });  

});