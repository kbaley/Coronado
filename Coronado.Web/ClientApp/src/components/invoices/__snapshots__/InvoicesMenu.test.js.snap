// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InvoicesMenu tests should render an InvoicesMenu 1`] = `
<li
  className=""
  role="presentation"
>
  <a
    href="/invoices"
    onClick={[Function]}
    onKeyDown={[Function]}
  >
    <span
      className="glyphicon glyphicon-list-alt"
    />
     Invoices
  </a>
</li>
`;
